File: update_review.py

import json
import random
import os

file_path = r'src/assets/review.json'

try:
    if not os.path.exists(file_path):
        print(f"Error: File not found at {file_path}")
        exit(1)

    with open(file_path, 'r', encoding='utf-8') as f:
        data = json.load(f)

    # Flatten the list if it contains nested lists
    flat_data = []
    def flatten(lst):
        for item in lst:
            if isinstance(item, list):
                flatten(item)
            else:
                flat_data.append(item)

    flatten(data)

    # Add data_cd to each item
    for item in flat_data:
        # Generate number between 1 and 422
        num = random.randint(1, 422)
        # Format as PTxxxx
        data_cd = f'PT{num:04d}'
        item['data_cd'] = data_cd

    # Write back to file
    with open(file_path, 'w', encoding='utf-8') as f:
        json.dump(flat_data, f, ensure_ascii=False, indent=2)

    print(f'Successfully updated {len(flat_data)} items.')

except Exception as e:
    print(f'Error: {e}')
    exit(1)

File: update_review.js
const fs = require('fs');
const path = require('path');

// Use relative path from project root
const filePath = path.join('src', 'assets', 'review.json');

try {
  console.log(`Checking file at: ${filePath}`);
  if (!fs.existsSync(filePath)) {
    console.error(`Error: File not found at ${filePath}`);
    process.exit(1);
  }

  const fileContent = fs.readFileSync(filePath, 'utf-8');
  let data = JSON.parse(fileContent);

  // Flatten the list if it contains nested lists
  let flatData = [];
  function flatten(lst) {
    for (const item of lst) {
      if (Array.isArray(item)) {
        flatten(item);
      } else {
        flatData.push(item);
      }
    }
  }

  if (Array.isArray(data)) {
    flatten(data);
  } else {
    flatData = [data]; 
  }

  console.log(`Found ${flatData.length} items.`);

  // Add data_cd to each item
  for (const item of flatData) {
    // Generate number between 1 and 422
    const num = Math.floor(Math.random() * 422) + 1;
    // Format as PTxxxx
    const data_cd = `PT${String(num).padStart(4, '0')}`;
    item['data_cd'] = data_cd;
  }

  // Write back to file
  fs.writeFileSync(filePath, JSON.stringify(flatData, null, 2), 'utf-8');

  console.log(`Successfully updated ${flatData.length} items.`);

} catch (e) {
  console.error(`Error: ${e.message}`);
  process.exit(1);
}